jiant/proj/main/modeling/model_setup.py

```diff
@@ -43,6 +43,11 @@ def setup_jiant_model(
         transformers_class_spec=transformers_class_spec, model_config_path=model_config_path,
     )
     encoder = get_encoder(model_arch=model_arch, ancestor_model=ancestor_model)
+    if model_arch == ModelArchitectures.XLM:
+        # Need to pass the id2lang and lang2id map to the encoder
+        xlm_specific_setup(
+            encoder=encoder, ancestor_model=ancestor_model, model_config_path=model_config_path,
+        )
```

Comment:
seems like this belongs in `create_taskmodel()`

Reply:
This affects and modifies the encoder in-place, whereas `create_taskmodel()` takes the encoder as a given, so it should be done during the overall model setup, and only once.

Reply:
I'm getting confused why we would ever need to pass around both `ancestor_model`: Configured model with pretraining heads and `encoder`: Bare pretrained model outputting raw hidden-states without a specific head on top since they both come from ancestor_model. Fair point with `create_task_config()`, I think a better place for `xlm_specific_setup` would be `get_ancestor_model` but renamed `create_ancestor_model`

Reply:
Makes sense. To clarify: we extract the encoder (e.g. `BertModel`) from the ancestor model (`BertForMaskedLM`). Maybe "pretrained transformer model"? The HF folks also mentioned there's actually a built-in method for this now.